consensus/spos/consensusMessageValidator_test.go

```diff
@@ -1,3 +1,455 @@
+	err := wrk.CheckMessageWithFinalInfoValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidPublicKeyBitmapSize))
+}
+
+func TestCheckMessageWithFinalInfoValidity_InvalidSignatureSize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{PubKeysBitmap: []byte("01"), AggregateSignature: []byte("0")}
+	err := wrk.CheckMessageWithFinalInfoValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
+}
+
+func TestCheckMessageWithFinalInfoValidity_InvalidLeaderSignatureSize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	sig := make([]byte, SignatureSize)
+	_, _ = rand.Read(sig)
+	cnsMsg := &consensus.Message{PubKeysBitmap: []byte("01"), AggregateSignature: sig, LeaderSignature: []byte("0")}
+	err := wrk.CheckMessageWithFinalInfoValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
+}
+
+func TestCheckMessageWithFinalInfoValidity_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	sig := make([]byte, SignatureSize)
+	_, _ = rand.Read(sig)
+	cnsMsg := &consensus.Message{PubKeysBitmap: []byte("01"), AggregateSignature: sig, LeaderSignature: sig}
+	err := wrk.CheckMessageWithFinalInfoValidity(cnsMsg)
+	assert.Nil(t, err)
+}
+
+func TestCheckMessageWithSignatureValidity_InvalidMessage(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{Body: []byte("body")}
+	err := wrk.CheckMessageWithSignatureValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessage))
+}
+
+func TestCheckMessageWithSignatureValidity_InvalidSignatureShareSize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{SignatureShare: []byte("0")}
+	err := wrk.CheckMessageWithSignatureValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
+}
+
+func TestCheckMessageWithSignatureValidity_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	sig := make([]byte, SignatureSize)
+	_, _ = rand.Read(sig)
+	cnsMsg := &consensus.Message{SignatureShare: sig}
+	err := wrk.CheckMessageWithSignatureValidity(cnsMsg)
+	assert.Nil(t, err)
+}
+
+func TestCheckMessageWithBlockHeaderValidity_InvalidMessage(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	cnsMsg := &consensus.Message{SignatureShare: []byte("0")}
+	err := wrk.CheckMessageWithBlockHeaderValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidMessage))
+}
+
+func TestCheckMessageWithBlockHeaderValidity_InvalidHeaderSize(t *testing.T) {
```

Comment:
You can also add a test for header size larger than 1MB exactly as you did for block body size

Reply:
done.